app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/components/return_charges.scala.html

```diff
@@ -15,49 +15,46 @@
-
+    chargeItem: uk.gov.hmrc.cgtpropertydisposalsfrontend.views.html.components.charge_item,
+    warning: uk.gov.hmrc.cgtpropertydisposalsfrontend.views.html.components.warning,
+    notice: uk.gov.hmrc.cgtpropertydisposalsfrontend.views.html.components.inset_panel
 )
 
-@(heading: String, returnSummary: ReturnSummary, isAgent: Boolean, isATrust: Boolean)(implicit messages: Messages)
-
-@key = @{ "viewReturn" }
-@userKey = @{
-    if (isAgent) ".agent"
-    else if (isATrust) ".trust"
-    else ""
-}
+@(heading: String, key : String, returnSummary: ReturnSummary)(implicit messages: Messages)
 
 <h2 class="heading-large">@heading</h2>
-@returnSummary.charges.filter(c => c.chargeType === UkResidentReturn || c.chargeType === NonUkResidentReturn).map { charge =>
-    @warning(messages(s"$key${userKey}.warning", TimeUtils.govDisplayFormat(charge.dueDate)))
-}
-
-@returnSummary.charges.filter(c => c.chargeType === UkResidentReturn || c.chargeType === NonUkResidentReturn).map { _ =>
-    @Html(messages(s"$key${userKey}.warning.subtext"))
-}
 
-<table class="borders payments-table view-return-charges">
-  <thead>
-    <tr style="vertical-align:bottom;">
-      <th scope="col">@messages(s"$key.charges.description")</th>
-      <th scope="col">@messages(s"$key.charges.payBy")</th>
-      <th scope="col">@messages(s"$key.charges.amount")</th>
-      <th scope="col">@messages(s"$key.charges.due")</th>
-      <th scope="col">@messages(s"$key.charges.status")</th>
-    </tr>
-  </thead>
-  <tbody id="returnPaymentDetails-@returnSummary.submissionId">
-    @{ returnSummary.charges.map(chargeItem(_)) }
-  </tbody>
-</table>
+    @if(returnSummary.isRecentlyAmended){
+        @warning(messages(s"${key}.amendment.warning"))
+        @notice(Html(messages(s"$key.amendment.notice")))
+    }else{
+        @returnSummary.charges.filter(c => c.chargeType === UkResidentReturn || c.chargeType === NonUkResidentReturn).map { charge =>
+            @warning(messages(s"${key}.warning", TimeUtils.govDisplayFormat(charge.dueDate)))
+        }
+
+        @returnSummary.charges.filter(c => c.chargeType === UkResidentReturn || c.chargeType === NonUkResidentReturn).map { _ =>
+            @Html(messages(s"${key}.warning.subtext"))
+        }
+        <table class="borders payments-table view-return-charges">
+            <thead>
+            <tr style="vertical-align:bottom;">
+                <th scope="col">@messages("viewReturn.charges.description")</th>
+                <th scope="col">@messages("viewReturn.charges.payBy")</th>
+                <th scope="col">@messages("viewReturn.charges.amount")</th>
+                <th scope="col">@messages("viewReturn.charges.due")</th>
+                <th scope="col">@messages("viewReturn.charges.status")</th>
+            </tr>
+            </thead>
+            <tbody id="returnPaymentDetails-@returnSummary.submissionId">
+            @{ returnSummary.charges.map(chargeItem(_))}
```

Comment:
you'll need to include the delta charge here as well if it exists. The description for it should be `Updated tax liability` if it exists.

Reply:
but also, if there is no delta charge the description of the main return charge type needs to be `Updated tax liability` as well for all amend returns

Reply:
Yeah, that's where I was passing in the delta object through but then ripped it all out. So looks like I was on the right lines there. Do I have to match the charge reference in the delta object with the charge reference on the Charge object and update the charge description?

Reply:
yeah, those are the two rows which should appear. The charge type's which need their messages updating for amend returns are `UkResidentReturn` and `NonUkResidentReturn`. In the message files we've currently got:
```
viewReturn.chargeType.UkResidentReturn=Tax payment
viewReturn.chargeType.NonUkResidentReturn=Tax payment
```
so you could just add some extra ones there like:
```
viewReturn.amend.chargeType.UkResidentReturn=Updated tax liability
viewReturn.amend.chargeType.NonUkResidentReturn=Updated tax liability
```
and make the template use those keys when needed. Then you could stick the delta charge row on the bottom

Reply:
can I also assume that given there is a delta then an amend has been done - ie I don't need to explicitly pass in a flag to the charge_item template to implement the logic as per the ticket; that's what I was doing before

Reply:
a delta charge implies an amend return has been done, but an amend return does not imply a delta charge will be raised. So you may need to pass in a separate flag cos the content needs to apply for all amend returns